cpp/src/arrow/util/formatting.h

```diff
@@ -40,12 +40,51 @@ namespace arrow {
 namespace internal {
 
 /// \brief The entry point for conversion to strings.
+///
+/// Specializations of FormatValueTraits for `ARROW_TYPE` must define:
+/// - A member type `value_type` which will be formatted.
+/// - A static member function `MaxSize(const ARROW_TYPE& t)` indicating the
+///   maximum number of characters which formatting a `value_type` may yield.
+/// - The static member function `Convert`, callable with signature
```

Comment:
Why not call it `Format`?

Reply:
I was following the convention of `ParseValueTraits`, which names the conjugate method `Convert`

Reply:
I see, fair enough.